### PR TITLE
docs: Update containerized-gadgets design document

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**
Update the design document with current implementation and learning from different PRs

### CODE REVIEW

File: docs/design/002-containerized-gadgets.md [hunk 1]
```diff
@@ -244,27 +316,65 @@ only by Inspektor Gadget.
...
-Inspektor Gadget automatically reads and sorts those maps according to the configuration provided by the user:
+Toppers are gadgets that print a list of elements sorted by specific parameters, like TCP
+connections by sent bytes. The gadget collects and stores the information on a hash map,and then
+Inspektor Gadget automatically reads and sorts those maps according to the configuration provided by
+the user:
 - Interval: How often to print the map contents
 - Sort by: Field within the event type to sort by
 
-#### BPF Iterators (a.k.a snapshotters)
+Gadget developers can use `GADGET_STATS_MAP` to indicate the map is used to store stats.
```
Comment (burak-ok) [inline docs/design/002-containerized-gadgets.md:328]:
```suggestion
Gadget developers can use `GADGET_STATS_MAP` to mark the map which stores the stats.
```

File: docs/design/002-containerized-gadgets.md [hunk 2]
```diff
@@ -244,27 +316,65 @@ only by Inspektor Gadget.
...
+```
+
+#### Snapshotters
 
 Programs of type `iter/` are automatically loaded and attached by Inspektor Gadget, then they are
 triggered one time and its output is parsed according to the BTF information provided by it.
 
-In order to have BTF information available, the programs have to define an unused variable with the
-type they output in the iterator:
+The `GADGET_SNAPSHOTTER` macro is available to indicate the type produced by a given snapshotter
```
Comment (burak-ok) [inline docs/design/002-containerized-gadgets.md:367]:
```suggestion
The `GADGET_SNAPSHOTTER` macro is available to indicate the struct type produced by a given snapshotter
```

**Review by burak-ok — APPROVED** [review]
This is going to be hard to keep this doc always up to date. But its needed and useful.
Also my english is also not the best, so take the suggestions with a grain of salt :D

Approve since this is always evolving

**Comment by mauriciovasquezbernal** [comment]
There are still big discussions about this design, hence I prefer closing this for now.

**Pull request closed without merging**